lib/galaxy/datatypes/molecules.py

```diff
@@ -674,6 +795,689 @@ def set_peek(self, dataset):
+                    pbc = block.get_pbc()
+                    try:
+                        p = 1 if pbc else 0
+                    except ValueError:  # pbc is an array
+                        p = 1 if pbc.any() else 0
+                    is_periodic.append(p)
+                    lattice_parameters.append(list(block.get_cell().cellpar()))
+            except Exception as e:
+                log.warning('%s, set_meta Exception during ASE metadata collection: %s', self, unicodify(e))
+                self.meta_error = True
+                return
+
+            dataset.metadata.number_of_molecules = len(ase_data)
+            dataset.metadata.atom_data = atom_data
+            dataset.metadata.number_of_atoms = [len(atoms) for atoms in dataset.metadata.atom_data]
+            dataset.metadata.chemical_formula = chemical_formula
+            dataset.metadata.is_periodic = is_periodic
+            dataset.metadata.lattice_parameters = list(lattice_parameters)
+        else:
+            # simple metadata
+            with open(dataset.file_name) as f:
+                xyz_lines = f.readlines()
+            try:
+                blocks = self.read_blocks(xyz_lines)
+            except (IndexError, TypeError, ValueError) as e:
+                # file does not follow XYZ structure
+                log.warning('%s, set_meta Exception during simple metadata collection: %s', self, unicodify(e))
+                self.meta_error = True
+                return
+
+            dataset.metadata.number_of_molecules = len(blocks)
+            dataset.metadata.atom_data = [block["atom_data"] for block in blocks]
+            dataset.metadata.number_of_atoms = [block["number_of_atoms"] for block in blocks]
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = get_file_peek(dataset.file_name)
+            dataset.info = self.get_dataset_info(dataset.metadata)
+            structure_string = "structure" if dataset.metadata.number_of_molecules == 1 else "structures"
+            dataset.blurb = f"XYZ file containing {dataset.metadata.number_of_molecules} {structure_string}"
+        else:
+            dataset.peek = "file does not exist"
+            dataset.blurb = "file purged from disk"
+
+    def get_dataset_info(self, metadata):
+        info = ""  # default to empty info
+
+        if self.meta_error:
+            info = "Error finding metadata. The file may be formatted incorrectly."
+        elif ase_io:
+            # enhanced info
+            info_list = []
+            if metadata.number_of_molecules == 1:
+                info_list.append(f"Chemical formula:\n{metadata.chemical_formula[0]}")
+                if metadata.is_periodic[0]:
+                    info_list.append("Periodic:\nYes")
+                    info_list.append(
+                        f"Lattice parameters in axis-angle format:\n{', '.join([str(round(x,2)) for x in metadata.lattice_parameters[0]])}"
```

Comment:
```suggestion
                        f"Lattice parameters in axis-angle format:\n{', '.join([str(round(x, 2)) for x in metadata.lattice_parameters[0]])}"
```

Reply:
I'm surprised this hasn't been flagged as a linting error.

Reply:
> I'm surprised this hasn't been flagged as a linting error.

It seems both flake8 and black don't check code inside f-strings placeholders.